src/page-multi-facility/ScenarioLibraryModal.tsx

```diff
@@ -165,11 +182,50 @@ const ScenarioLibraryModal: React.FC<Props> = ({ trigger }) => {
   const [showDeleteModal, setShowDeleteModal] = useState(false);
   const rejectionToast = useRejectionToast();
 
+  const sortScenarios = (list: any) => {
+    if (Array.isArray(list)) {
+      const baseline: any = [];
+      const nonBaseline: any = [];
+      list.forEach((scenario: any) => {
+        if (scenario.baseline) baseline.push(scenario);
+        else nonBaseline.push(scenario);
+      });
+
+      baseline.sort((a: any, b: any) =>
+        a.name.toLowerCase() > b.name.toLowerCase() ? 1 : -1,
+      );
+      nonBaseline.sort((a: any, b: any) =>
+        a.name.toLowerCase() > b.name.toLowerCase() ? 1 : -1,
+      );
+
+      return [...baseline, ...nonBaseline];
+    } else {
+      return list;
+    }
+  };
+
   async function fetchScenarios() {
     const scenariosData = await getScenarios();
     if (scenariosData) {
-      setScenarios({
-        data: scenariosData,
+      const oScenarios = [] as Scenario[];
+      const sScenarios = [] as Scenario[];
+      const userId = currrentUserId();
+
+      scenariosData.forEach((scenario) => {
+        if (userId && scenario.roles[userId] === "owner") {
+          oScenarios.push(scenario);
+        } else {
+          sScenarios.push(scenario);
+        }
+      });
+
+      setOwnedScenarios({
+        data: oScenarios,
+        loading: false,
+      });
+
+      setSharedScenarios({
+        data: sortScenarios(sScenarios),
```

Comment:
Assuming that insertion order is preserved with `push`, I _think_ we can just keep all of the sorting logic in the `getScenarios` function and just have this function tease apart the `owned` and `shared` scenarios without having to resort them here.

Reply:
Yes I second this; the incoming sort order of `scenariosData` should be preserved by this code

Reply:
suggestion implemented!